File: trunk/main/src/com/seovic/coherence/loader/CsvToCoherence.java
/*
 * Copyright 2009 Aleksandar Seovic
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.seovic.coherence.loader;


import com.seovic.coherence.loader.source.CsvSource;
import com.seovic.coherence.loader.target.CoherenceCacheTarget;

import com.tangosol.net.NamedCache;
import com.tangosol.net.CacheFactory;

import java.io.Reader;
import java.io.FileReader;

import org.springframework.core.io.Resource;


/**
 * Convenience class that loads data from the CSV file into Coherence cache
 * using default settings.
 *
 * @author Aleksandar Seovic  2009.09.29
 */
public class CsvToCoherence
        extends AbstractDelegatingLoader
    {
    // ---- constructors ----------------------------------------------------

    /**
     * Construct CsvToCoherence loader instance.
     *
     * @param csvFile    CSV file to read items from
     * @param cache      Coherence cache to import objects into
     * @param itemClass  target item class
     */
    public CsvToCoherence(String csvFile, NamedCache cache, Class itemClass)
        {
        this(getResource(csvFile), cache, itemClass);
        }

    /**
     * Construct CsvToCoherence loader instance.
     *
     * @param csvFile    CSV file to read items from
     * @param cache      Coherence cache to import objects into
     * @param itemClass  target item class
     */
    public CsvToCoherence(Resource csvFile, NamedCache cache, Class itemClass)
        {
        this(createResourceReader(csvFile), cache, itemClass);
        }

    /**
     * Construct CsvToCoherence loader instance.
     *
     * @param csvReader  CSV file reader
     * @param cache      Coherence cache to import objects into
     * @param itemClass  target item class
     */
    public CsvToCoherence(Reader csvReader, NamedCache cache, Class itemClass)
        {
        Source source = new CsvSource(csvReader);
        Target target = new CoherenceCacheTarget(cache, itemClass);
        setLoader(new DefaultLoader(source, target));
        }


    // ---- main method -----------------------------------------------------

    public static void main(String[] args)
            throws Exception
        {
        if (args.length < 3)
            {
            System.out.println("Usage: java com.seovic.coherence.loader.CsvToCoherence <csvFile> <cacheName> <itemClass>");
            System.exit(0);
            }

        FileReader csvReader = new FileReader(args[0]);
        NamedCache cache     = CacheFactory.getCache(args[1]);
        Class      itemClass = Class.forName(args[2]);

        new CsvToCoherence(csvReader, cache, itemClass).load();
        }
    }

File: main.net/src/Coherence.Tools/Coherence/Loader/CsvToCoherence.cs
using System;
using System.IO;
using Seovic.Loader;
using Seovic.Loader.Source;
using Spring.Core.IO;
using Tangosol.Net;

namespace Seovic.Coherence.Loader
{
    /// <summary>
    /// Convenience class that loads data from the CSV file into Coherence cache
    /// using default settings.
    /// </summary>
    /// <author>Aleksandar Seovic  2009.09.29</author>
    /// <author>Ivan Cikic  2009.10.31</author>
    public class CsvToCoherence : AbstractDelegatingLoader
    {
        #region Constructors

        /// <summary>
        /// Construct CsvToCoherence loader instance.
        /// </summary>
        /// <param name="csvFile">CSV file to read items from</param>
        /// <param name="cache">Coherence cache to import objects into</param>
        /// <param name="itemType">target item class</param>
        public CsvToCoherence(String csvFile, INamedCache cache, Type itemType) 
            : this(GetResource(csvFile), cache, itemType)
        {
        }

        /// <summary>
        /// Construct CsvToCoherence loader instance.
        /// </summary>
        /// <param name="csvFile">CSV file to read items from</param>
        /// <param name="cache">Coherence cache to import objects into</param>
        /// <param name="itemType">target item class</param>
        public CsvToCoherence(IResource csvFile, INamedCache cache, Type itemType) 
            : this(CreateResourceReader(csvFile), cache, itemType)
        {
        }

        /// <summary>
        /// Construct CsvToCoherence loader instance.
        /// </summary>
        /// <param name="csvReader">CSV file reader</param>
        /// <param name="cache">Coherence cache to import objects into</param>
        /// <param name="itemType">target item class</param>
        public CsvToCoherence(TextReader csvReader, INamedCache cache, Type itemType)
        {
            ISource source = new CsvSource(csvReader);
            ITarget target = new CoherenceCacheTarget(cache, itemType);
            Loader = new DefaultLoader(source, target);
        }


        #endregion

        #region Main method

        /// <summary>
        /// Main method.
        /// </summary>
        /// <param name="args">Command line arguments.</param>
        public static void Main(string[] args)
        {
            if (args.Length < 3)
            {
                Console.WriteLine("Usage: CsvToCoherence <csvFile> <cacheName> <itemClass>");
            }

            TextReader  csvReader = new StreamReader(args[0]);
            INamedCache cache     = CacheFactory.GetCache(args[1]);
            Type        itemType  = Type.GetType(args[2]);

            new CsvToCoherence(csvReader, cache, itemType).Load();
        }

        #endregion
    }
}
